src/main/java/com/limechain/trie/TrieAccessor.java

```diff
@@ -93,17 +90,11 @@ public Optional<byte[]> findMerkleValue(Nibbles key) {
      * Deletes the value associated with the given key from the trie.
      *
      * @param key The key to delete.
-     * @return    True if the operation was successful, false otherwise.
+     * @return True if the operation was successful, false otherwise.
      */
     @Override
     public boolean delete(Nibbles key) {
-        Entry<NodeData> node = initialTrie.node(key);
-        if (!(node instanceof Vacant<NodeData>)) {
-            NodeHandle<NodeData> nodeHandle = node.asNodeHandle();
-            initialTrie.clearNodeValue(nodeHandle.getNodeIndex());
-            return true;
-        }
-        return false;
+        return initialTrie.deleteNodeAt(key);
```

Comment:
Whilst `deleteNodeAt` seems an appropriate name for what the method does, but we're missing the check here for whether it's an actual storage node at this key. With the current implementation we're able to accidentally remove a branch node with no storage value if it's full key happens to coincide with the one requested.

This could either be delegated to the TrieStructure as a responsibility (i.e. an actual `clearStorageValue` method which does the check), but I'd prefer to leave that at the caller as to not introduce further responsibilities to the `TrieStructure` class.